dag_in_context/src/interval_analysis.egg

```diff
@@ -63,32 +63,36 @@
        (= (IntB x) (lo-bound expr))
        (= (IntB x) (hi-bound expr))
        (HasArgType expr ty)
+       (ContextOf expr ctx)
       )
-      ((union expr (Const (Int x) ty)))
+      ((union expr (InContext ctx (Const (Int x) ty))))
       :ruleset interval-analysis)
 
 (rule (
        (= (BoolB x) (lo-bound expr))
        (= (BoolB x) (hi-bound expr))
        (HasArgType expr ty)
+       (ContextOf expr ctx)
       )
-      ((union expr (Const (Bool x) ty)))
+      ((union expr (InContext ctx (Const (Bool x) ty))))
```

Comment:
Oh interesting, I didn't realize this was a problem. What if `expr` isn't in a context?

Reply:
Everything has a context! I'm writing these invariants down in the code and paper soontm